### PR TITLE
Implement 'delete endpoint' command

### PR DESCRIPTION
**Pull request by axbarsan**
Fixes https://github.com/giantswarm/giantswarm/issues/8442

### CODE REVIEW

File: commands/delete/command.go [hunk 1]
```diff
@@ -12,11 +13,12 @@ var (
 	Command = &cobra.Command{
 		Use:   "delete",
 		Short: "Delete things",
-		Long:  `Lets you delete a cluster or node pool`,
+		Long:  `Lets you delete a cluster, a node pool or an API endpoint`,
```
Comment (oponder) [inline commands/delete/command.go]:
```suggestion
		Long:  `Lets you delete a cluster, a node pool, or an API endpoint`,
```

Up in these woods we use the oxford comma :P

🇬🇧

File: commands/delete/endpoint/command_test.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+			}
+			if diff := cmp.Diff(tc.resultingArgs, args); diff != "" {
+				t.Errorf("Case %d - Resulting args unequal. (-expected +got):\n%s", i, diff)
+			}
+		})
+	}
+}
+
+// TestDeleteEndpointSuccess runs test case that are supposed to succeed
+func TestDeleteEndpointSuccess(t *testing.T) {
```
Comment (oponder) [inline commands/delete/endpoint/command_test.go]: Should we do a test of printResult as well? Just noticing the decrease in coverage.

**Review by oponder — CHANGES_REQUESTED** [review]
Thanks! Just have a grammar suggestion and a testing question.

File: commands/delete/endpoint/command_test.go (hunk 2 above)
Comment (axbarsan) [reply commands/delete/endpoint/command_test.go]: Yes, that's on its way 😬

**Review by oponder — APPROVED** [review]
LGTM IIWFY!

**Comment by marians** [comment]
I am wondering whether the confirmation is really needed.

**PRO**: We do it with all other deletions, so users may expect it.

**CON**: Deleting an endpoint removes the entry and auth token from the config. It is not destroying any actual data in the primary sense, like deleting a cluster may.

But when in doubt, it's probably fine to have the confirmation.

**Comment by axbarsan** [comment]
As you said, we have confirmations in all of our deletions, so the user will expect it to be here, also.
Deleting it accidentally would not make you lose valuable data, but will make you lose time, which is equally important 👍

File: commands/delete/endpoint/command.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+		case errors.IsEndpointMissingError(err):
+			headline = "No API endpoint specified"
+			subtext = "See --help for usage details."
+		case errors.IsCouldNotDeleteEndpointError(err):
+			headline = "The API endpoint could not be deleted."
+			subtext = "You might try again in a few moments. If that doesn't work, please contact the Giant Swarm support team."
+			subtext += "Sorry for the inconvenience!"
+		default:
+			headline = err.Error()
+		}
```
Comment (marians) [inline commands/delete/endpoint/command.go]: Here in `printValidation` we only have to handle the errors that can occur in `validatePreconditions`.

`errors.IsCouldNotDeleteEndpointError` is not returned by `validatePreconditions` so should not be handled here. Instead it has to be handled in the `printResult` function. Also `client.HandleErrors(err)` isn't needed (but also doesn't hurt) here, as there is no client usage in `validatePreconditions`.

**Review by marians — CHANGES_REQUESTED** [review]

**Comment by marians** [comment]
Confirmation is fine with me, however I think having to type some string is over-cautious.

```
./gsctl delete endpoint gauss
Do you really want to delete API endpoint 'gauss'? Please type the endpoint name to confirm:
```

A simple n/y (case insensitive) should do here.

```
./gsctl delete endpoint gauss
Do you really want to delete API endpoint 'gauss'? [y/N]:
```

"N" is capitalized to indicate that it is assumed to be the default when nothing is entered.

**Comment by axbarsan** [comment]
> Confirmation is fine with me, however I think having to type some string is over-cautious.
> 
> ```
> ./gsctl delete endpoint gauss
> Do you really want to delete API endpoint 'gauss'? Please type the endpoint name to confirm:
> ```
> 
> A simple n/y (case insensitive) should do here.
> 
> ```
> ./gsctl delete endpoint gauss
> Do you really want to delete API endpoint 'gauss'? [y/N]:
> ```
> 
> "N" is capitalized to indicate that it is assumed to be the default when nothing is entered.

I've added this functionality to all the `yes/no` confirms

**Comment by marians** [comment]
Wording nitpick:

    The API endpoint '<endpoint-or-alias>' deleted successfully.

could be changed to

    The API endpoint '<endpoint-or-alias>' has been deleted.

Found a little UX problem we might want to work around in this PR.

When deleting the currently selected endpoint, and then issuing commands like `gsctl list clusters`, this error occurs:

```
You are not logged in.
Use 'gsctl login' to login or '--auth-token' to pass a valid auth token.
```

The error message is misleading and I created https://github.com/giantswarm/giantswarm/issues/8843 for that.

However, we could do more here in the context of the `delete endpoint` command to prevent users from running into this. When deleting the current endpoint, I suggest to show this additional output:

```
The API endpoint '<endpoint-or-alias>' has been deleted.

Note: You have no API endpoint selected now. Use 'gsctl select endpoint' to select one.
```

File: confirm/confirm.go [hunk 4]
```diff
@@ -20,17 +20,19 @@ func Ask(s string) bool {
...
 
 		response, err := reader.ReadString('\n')
 		if err != nil {
 			log.Fatal(microerror.Mask(err))
 		}
 
 		switch strings.ToLower(strings.TrimSpace(response)) {
-		case "y", "yes":
+		case "y", "yes", "Y", "YES", "Yes":
 			return true
```
Comment (marians) [inline confirm/confirm.go]: As you convert the string to be tested to lower case, there should be node need to test `"Y", "YES", "Yes"` , right?

**Review by marians — APPROVED** [review]

File: confirm/confirm.go (hunk 4 above)
Comment (axbarsan) [reply confirm/confirm.go]: You're right, I didn't catch that. Thank you 👍

**Pull request merged**